molter/help.py

```diff
@@ -0,0 +1,187 @@
+    show_hidden: bool = attrs.field(default=False, kw_only=True)
+    """Should hidden commands be shown"""
+    show_disabled: bool = attrs.field(default=False, kw_only=True)
+    """Should disabled commands be shown"""
+    run_checks: bool = attrs.field(default=False, kw_only=True)
+    """Should only commands that's checks pass be shown"""
+    show_self: bool = attrs.field(default=False, kw_only=True)
+    """Should this command be shown in the help message"""
+    show_usage: bool = attrs.field(default=False, kw_only=True)
+    """Should usage for commands be shown"""
+    show_aliases: bool = attrs.field(default=False, kw_only=True)
+    """Should aliases for commands be shown"""
+    show_prefix: bool = attrs.field(default=False, kw_only=True)
+    """Should the prefix be shown"""
+    embed_color: Color = attrs.field(default=BrandColors.BLURPLE, kw_only=True)
+    """The colour to show in the Embeds"""
+
+    embed_title: str = attrs.field(default="{username} Help Command", kw_only=True)
+    """The title to use in the embed. {username} will be replaced by the client's username"""
+    not_found_message: str = attrs.field(default="Sorry! No command called `{cmd_name}` was found.", kw_only=True)
+    """The message to send when a command was not found. {cmd_name} will be replaced by the requested command."""
+
+    _client: "Snake" = attrs.field()
+    _cmd: molter.MolterCommand | None = attrs.field(init=False, default=None)
+
+    def __attrs_post_init__(self) -> None:
+        if not self._cmd:
+            self._cmd = self._callback  # type: ignore
+
+    def register(self) -> None:
+        """Register the help command in dis-snek"""
+        if not isinstance(self._cmd.callback, functools.partial):
+            # prevent wrap-nesting
+            self._cmd.callback = functools.partial(self._cmd.callback, self)
+
+        # replace existing help command if found
+        if "help" in self._client.commands:
+            log.warning("Replacing existing help command.")
+            del self._client.commands["help"]
+
+        self._client.add_message_command(self._cmd)  # type: ignore
+
+    async def send_help(self, ctx: MessageContext, cmd_name: str | None) -> None:
+        """
+        Send a help message to the given context.
+
+        args:
+            ctx: The context to use.
+            cmd_name: An optional command name to send help for.
+        """
+        await self._callback.callback(ctx, cmd_name)  # type: ignore
+
+    @molter.msg_command(name="help")
+    async def _callback(self, ctx: MessageContext, *, cmd_name: str = None) -> None:
+        if cmd_name:
+            return await self._help_specific(ctx, cmd_name)
+        await self._help_list(ctx)
+
+    async def _help_list(self, ctx: MessageContext) -> None:
+        cmds = await self._gather(ctx)
+
+        output = []
+        for cmd in cmds.values():
+            _temp = self._generate_command_string(cmd, ctx)
+            _temp += f"\n{cmd.brief}"
```

Comment:
`cmd.brief` has the chance to be `None` if they do not provide docstrings. I feel like *something* should be put here in its place if so (like "No description provided." or the like), though it *may* be worth letting the user customize what should be put as a placeholder here - I'll let you decide.